src/frontend/js/hooks/useUnionResource/index.ts

```diff
@@ -132,7 +149,7 @@ const useUnionResource = <
       setIsSyncing(true);
       fetchNewPage();
     }
-  }, [cursor]);
+  }, [cursor, stack.length]); // stack.length is added in the dependency array to force a new fetch on reset.
```

Comment:
will this double the amount of execution of this useEffect  ? 
* one when we set cursor to "stack + per_page"
* one when integrity is ok and that we set stack with new fetched entites (l.135)

I see two other way to do it:
* in `reset`, to use `setCursor(0)` and add a new `useEffect` that will set back cursor to `per_page` when `cursor === 0`.
* have `useQueryKeyInvalidateListener` return the `refershFlag` and use this `refreshFlag` there to trigger a new sync.

Reply:
Yes but the second will not be triggering any additional render.

Using `useQueryKeyInvalidateListener` will not be possible as we want this hook to be optional, and setting cursor to 0 looks a bit odd semantically in my opinion. What really reflects the reality is that we simply trash the stack, so as this hook must guarantee a correct output, we must take into account the mutation of `stack` IMO